tests/common-tests.h

```diff
@@ -172,6 +172,66 @@ std::ostream& operator<<(std::ostream& ostr, const std::vector<MyOptionalString>
     return ostr;
 }
 
+// The next classes are used for DDL testing
+template<typename T>
+struct DdlTester {
+    typedef T val_type;
+    DdlTester(soci::db_type type, T val, bool expectNull)
+        : inType(type), inVal(val), expectNull(expectNull) {}
+
+    soci::db_type inType;
+    T inVal;
+    bool expectNull;
+    soci::db_type outType;
+    T outVal;
+    bool isNull = false;
+};
+
+template<typename T>
+void check(DdlTester<T> const &val)
+{
+    CHECK(val.inType == val.outType);
+    CHECK(val.expectNull == val.isNull);
+    if (val.expectNull)
+        return;
+    CHECK(val.inVal == val.outVal);
+}
+
+template<>
+void check(DdlTester<double> const &val)
+{
+    CHECK(val.inType == val.outType);
+    CHECK(val.expectNull == val.isNull);
+    if (val.expectNull)
+        return;
+    CHECK(std::fpclassify(val.inVal) == std::fpclassify(val.outVal));
+    if (std::isnormal(val.inVal) && std::isnormal(val.outVal))
+        CHECK_THAT(val.inVal, Catch::Matchers::WithinRel(val.outVal));
```

Comment:
I suggest extracting this and the `CHECK(val.inVal == val.outVal);` line from above into a separate `DdlTester::check_roundtrip()` function and then removing this `check()` because the first 4 lines are identical for all types and should be done directly in `test_limit()`.